### DISCUSSION TITLE
Widget|Custom for Health Markers

### DISCUSSION THREAD

**Discussion by RicardoPhilippe** [Backupiseasy/ThreatPlates#458] [Ideas]
Here is a example: https://wago.io/yBzHj6t19

Hey man, I would like help in Threat Plates improvements, may you share your tool sets (IDEs, debug tools, etc), to help me to start in this world? =)
I really does not know how to start and what is more adopted in addon development.

@Backupiseasy @suicidalkatt

**Comment by Backupiseasy** [comment]
Hi @RicardoPhilippe, sure, I can list here what I am currently using for development. I need to think about what should be most important, so it might take me a day or two.

**Comment by Backupiseasy** [comment]
So, here's what I am using for addon development:

- VS Code as IDE with several extensions; the most important ones should be EmmyLua, Better Comments, colorize, External libraries, indent-rainbow, Project Manager (nice to add windows for my several installations of WoW)
- I added the WoW UI sourcecode https://github.com/Gethe/wow-ui-source as well as other addons to the Explorer (which External libraries).  I often look into this code to find out how something works. 
- GitKraken as Git client
- Webpages like WoW.tools and Globe Check
- For debugging, I really have no good solution in game. I mostly use print messages as well as Lua browser (an addon) to look into tables
- I also setup automation with GitHub Actions for packaging (using the BigWigs packager) and updating localization (from CurseForge)

I guess that should be the most important stuff. 

I started addon development by actually looking into the source code of existing addons. As I already know several programming languages,  learning Lua wasn't that difficult. I still look up some stuff (like metatables) in tutorials when I don't know something. Also, the WoWUIDev discord helps a lot in understanding WoW Lua better.
